controllers/reconcile.go

```diff
@@ -605,78 +605,45 @@ type ObjectStoreBucket struct {
 	TLS       *bool
 }
 
-// ExtractBucketConfigFromSecret extracts ObjectStoreBucket configuration for a specific bucket name
-// from a given Kubernetes Secret. It searches for the bucket name either directly
-// in the secret's Data map under the "bucket" key, or within the
-// "clowder/bucket-names" annotation (if it's a comma-separated list).
-//
-// It returns the ObjectStoreBucket configuration and an error if the secret data is
-// invalid or incomplete for the specified bucket, or if the bucket name isn't found.
-func ExtractBucketConfigFromSecret(secrets []v1.Secret, targetBucketName string) (*ObjectStoreBucket, error) {
+// ExtractBucketConfigFromSecretByName extracts ObjectStoreBucket configuration from the provided k8 secret name
+func ExtractBucketConfigFromSecretByName(secrets []v1.Secret, targetSecretName string) (*ObjectStoreBucket, error) {
 	if len(secrets) == 0 {
-		return nil, fmt.Errorf("no secrets provided to search for bucket '%s'", targetBucketName)
+		return nil, fmt.Errorf("no secrets provided to search for secret '%s'", targetSecretName)
 	}
 
-	clowderBucketNamesAnnotation := "clowder/bucket-names"
-
 	for _, secret := range secrets {
-		currentSecret := secret
-
-		// Check if essential credential keys are present in the current secret's Data
-		requiredCredentialKeys := []string{
-			"aws_access_key_id",
-			"aws_secret_access_key",
-		}
-
-		credentialsPresent := true
-		for _, key := range requiredCredentialKeys {
-			if _, ok := currentSecret.Data[key]; !ok {
-				credentialsPresent = false
-				break
+		// Check if the current secret's metadata name matches the target name.
+		if secret.Name == targetSecretName {
+			// Found the target secret, now validate and extract the data.
+			currentSecret := secret
+
+			// Check for essential credential keys.
+			requiredCredentialKeys := []string{
+				"bucket",
+				"aws_region",
+				"endpoint",
+				"aws_access_key_id",
+				"aws_secret_access_key",
 			}
-		}
-		if !credentialsPresent {
-			continue
-		}
 
-		bucketNameFoundInThisSecret := false
-
-		// Check for "bucket" key in secret.Data
-		if secretBucketData, ok := currentSecret.Data["bucket"]; ok {
-			if string(secretBucketData) == targetBucketName {
-				bucketNameFoundInThisSecret = true
-			}
-		}
-
-		// If not found in Data, check "clowder/bucket-names" annotation
-		if !bucketNameFoundInThisSecret {
-			if annoValue, ok := currentSecret.Annotations[clowderBucketNamesAnnotation]; ok {
-				annotatedBucketNames := strings.Split(annoValue, ",")
-
-				bucketNameFoundInThisSecret = slices.ContainsFunc(annotatedBucketNames, func(name string) bool {
-					return strings.TrimSpace(name) == targetBucketName
-				})
+			for _, key := range requiredCredentialKeys {
```

Comment:
@tahmidefaz we can specify each one of these as separate environment variables that have values derived from the secret via app-interface. Let me get you an example